service/qos/qos_common.cc

```diff
@@ -183,8 +185,60 @@ future<service_levels_info> get_service_level(cql3::query_processor& qp, std::st
             logger.warn("Failed to fetch data for service level {}: {}", service_level_name, std::current_exception());
         }
     }
-    
+
     co_return service_levels;
 }
 
+static std::optional<sstring> describe_service_level(std::string_view sl_name, const service_level_options& sl_opts) {
+    using slo = service_level_options;
+
+    utils::small_vector<sstring, 2> opts{};
+
+    if (std::holds_alternative<slo::delete_marker>(sl_opts.timeout)) {
+        return std::nullopt;
+    }
+    if (auto maybe_timeout = std::get_if<lowres_clock::duration>(&sl_opts.timeout)) {
+        opts.push_back(seastar::format("timeout = {}", *maybe_timeout));
+    }
+
+    switch (sl_opts.workload) {
+        case slo::workload_type::delete_marker:
+            // TODO: Is it possible that this is set to `delete_marker`, but `sl_opts.timeout` is not?
```

Comment:
I think it's not possible

Reply:
`on_internal_error` added in relevant places